json-schema-serializer/src/main/java/io/confluent/kafka/formatter/json/JsonSchemaMessageReader.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+
+package io.confluent.kafka.formatter.json;
+
+import kafka.common.KafkaException;
+import kafka.common.MessageReader;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.serialization.Serializer;
+import org.everit.json.schema.ValidationException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.json.JsonSchema;
+import io.confluent.kafka.schemaregistry.json.JsonSchemaProvider;
+import io.confluent.kafka.schemaregistry.json.jackson.Jackson;
+import io.confluent.kafka.serializers.AbstractKafkaSchemaSerDeConfig;
+import io.confluent.kafka.serializers.json.AbstractKafkaJsonSchemaSerializer;
+
+/**
+ * Example
+ * To use JsonSchemaMessageReader, first make sure that Zookeeper, Kafka and schema registry
+ * server are
+ * all started. Second, make sure the jar for JsonSchemaMessageReader and its dependencies are
+ * included
+ * in the classpath of kafka-console-producer.sh. Then run the following
+ * command.
+ *
+ * <p>1. Send JSON Schema string as value. (make sure there is no space in the schema string)
+ * bin/kafka-console-producer.sh --broker-list localhost:9092 --topic t1 \
+ * --line-reader io.confluent.kafka.formatter.JsonSchemaMessageReader \
+ * --property schema.registry.url=http://localhost:8081 \
+ * --property value.schema='{"type":"string"}'
```

Comment:
Comment should be probably be updated with JSONSchema

Reply:
Ack